### DISCUSSION TITLE
[Bug] logback don't print trace ID in vert.x application

### DISCUSSION THREAD

**Discussion by nnzbz** [apache/skywalking#10958] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

Java Agent (apache/skywalking-java)

### What happened

Trace ID is 'N/A'
<img width="974" alt="image" src="https://github.com/apache/skywalking/assets/3138908/bc812457-44ff-4a71-83b4-e355e5e6bd9c">

Environment: Vertx v4.4.3, logback v1.4.8, apm-toolkit-logback-1.x v8.16.0

this is my main src

- pom.xml
<img width="589" alt="image" src="https://github.com/apache/skywalking/assets/3138908/ad042da4-17d8-4e2a-90f7-49d81814f092">

- MainVerticle.java
<img width="600" alt="image" src="https://github.com/apache/skywalking/assets/3138908/791e14ba-dbc8-4830-95a6-1cf403ee3f4f">

### What you expected to happen

log print trace id is not 'N/A'

### How to reproduce

test project in https://github.com/bugsrc/skywalking-vertx-bugsrc

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
There are tons of discussion about printing trace ID. 
It is never guaranteed, but bases on current thread context.
React style(vertx too) has a lot of thread context switching, so, not fully covered.

**Reply by wu-sheng** [reply]
I move this to discussion. It is not a thing must have. 
Leave to people interested in this topic only.
